src/test/test/opinion/rules/ShouldRuleTest.java

```diff
@@ -0,0 +1,72 @@
+package opinion.rules;
+
+import opinion.rules.impls.ShouldRule;
+import opinion.sentences.Sentence;
+import org.junit.Assert;
+import org.junit.Test;
+import proto.RuleResultOuterClass;
+
+import java.util.List;
+
+
+public class ShouldRuleTest {
+
+    private RuleEngine ruleEngine = new RuleEngine(getTestRuleList());
+    private SentenceRule rule = RuleFactory.getRule("should");
```

Comment:
Please do not include `RuleEngine` or `RuleFactory` in this test. Not only are they not necessary (you can get an instance of `ShouldRule` w/o the `RuleFactory`, and you can apply rule without `RuleEngine`), but using them as the interface to test the logic of `ShouldRule` will make finding bugs difficult.